From ce933bdba2839ee9dd395e2bda7e7889766c8516 Mon Sep 17 00:00:00 2001
From: Zhang Xiao <xiao.zhang@windriver.com>
Date: Wed, 27 Aug 2014 13:34:15 +0800
Subject: [PATCH] yaffs: Avoid setting any ACL releated xattr

YAFFS doesn't sopport ACL yet, it must refuse any related settings.

Signed-off-by: Zhang Xiao <xiao.zhang@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 fs/yaffs2/yaffs_vfs.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/fs/yaffs2/yaffs_vfs.c b/fs/yaffs2/yaffs_vfs.c
index f4c5ba95d51d..611ef43d9825 100644
--- a/fs/yaffs2/yaffs_vfs.c
+++ b/fs/yaffs2/yaffs_vfs.c
@@ -970,6 +970,12 @@ static int yaffs_setxattr(struct dentry *dentry, const char *name,
 
 	yaffs_trace(YAFFS_TRACE_OS, "yaffs_setxattr of object %d", obj->obj_id);
 
+	/* Currently we don't support posix ACL so never accept any settings
+	 * start with "system.posix_acl_".
+	 */
+	if (strncmp(name, "system.posix_acl_", 17))
+		error = -EOPNOTSUPP;
+
 	if (error == 0) {
 		int result;
 		dev = obj->my_dev;
-- 
2.39.2

